### DISCUSSION TITLE
Missing package product 'Web3Modal' when using 1.6.12 SDK from Xcode SPM

### DISCUSSION THREAD

**Discussion by itsallmememe** [WalletConnect/walletconnect-monorepo#3122] [Developer Support]
**Describe the bug**
I cannot use the 1.6.12 SDK via SPM because I get the error "Missing package product 'Web3Modal'" when trying to build. Where is this dependency? How do I install it?

**SDK Version**

Client: Swift
Version 1.6.12
Xcode 14.2

**To Reproduce**

- Start a new project
- Add WalletConnect Swift lib via SPM
- Try to build
- Get error "Missing package product 'Web3Modal'"

**Expected behavior**
Using SPM the WalletConnect sdk should build automatically, and not require any further dependencies to be installed by the engineer.

Screenshots

![](https://user-images.githubusercontent.com/688771/253101022-ff68bf53-a726-46b3-991b-9a18633f7b0e.png)
![](https://user-images.githubusercontent.com/688771/253101090-e66aec15-414d-4762-b2da-e8ff31d2612f.png)

**Device** (please complete the following information):

Device: Apple M1 MackBook Pro
OS: MacOS 13.3.1
Additional context

**Comment by radeknovis** [comment]
Closing since the issue raised from this discussion got resolved

Posting solution here as well

>We recently renamed the `Web3Modal` target to `WalletConnectModal`, so I'm assuming it might be causing this issue. I tried adding it like this to a freshly created project, and it works. 
> 
> ![Screenshot 2023-07-13 at 13 08 34](https://github.com/WalletConnect/WalletConnectSwiftV2/assets/17084142/1a802c71-12dd-4798-a009-074b5098f7b4)
> 
> Purging the SPM cache and fetching the package again should help, if not, try removing the dependency and adding it  again also making sure that your target is linking correct library and not `Web3Modal`
> 
> ![Screenshot 2023-07-13 at 13 30 23](https://github.com/WalletConnect/WalletConnectSwiftV2/assets/17084142/ea45025b-5820-4b62-9b85-e3fd888f2ecc)

**Comment by itsallmememe** [comment]
👍
